stdlib/public/core/SwiftNativeNSArray.swift

```diff
@@ -129,6 +141,189 @@ extension __SwiftNativeNSArrayWithContiguousStorage: _NSArrayCore {
   internal func copy(with _: _SwiftNSZone?) -> AnyObject {
     return self
   }
+  
+  @objc(mutableCopyWithZone:)
+  internal func mutableCopy(with _: _SwiftNSZone?) -> AnyObject {
+    let arr = Array<AnyObject(_ContiguousArrayBuffer(self))
+    return _SwiftNSMutableArray(arr)
+  }
+  
+  @objc(indexOfObjectIdenticalTo:)
+  internal func index(ofObjectIdenticalTo object: AnyObject) -> Int {
+    let arr = Array<AnyObject(_ContiguousArrayBuffer(self))
+    return arr.firstIndex { $0 === object } ?? NSNotFound
+  }
+}
+
+@_fixed_layout
+@usableFromInline
+@objc internal final class _SwiftNSMutableArray :
+  _SwiftNativeNSMutableArray, _NSArrayCore
+{
+  internal var contents: [AnyObject]
+
+  internal init(_ array: [AnyObject]) {
+    contents = array
+    super.init()
+  }
+  
+  @objc internal var count: Int {
+    return contents.count
+  }
+  
+  @objc(objectAtIndexedSubscript:)
+  internal func objectAtSubscript(_ index: Int) -> AnyObject {
+    //TODO: exception instead of precondition, once that's possible
+    return contents[index]
+  }
+
+  @objc(objectAtIndex:)
+  internal func objectAt(_ index: Int) -> AnyObject {
+    //TODO: exception instead of precondition, once that's possible
+    return contents[index]
+  }
+
+  @objc internal func getObjects(
+    _ aBuffer: UnsafeMutablePointer<AnyObject>, range: _SwiftNSRange
+  ) {
+    return contents.withContiguousStorageIfAvailable { objects in
+      //TODO: exceptions instead of preconditions, once that's possible
+
+      _precondition(
+        _isValidArrayIndex(range.location, count: objects.count),
+        "Array index out of range")
+
+      _precondition(
+        _isValidArrayIndex(
+          range.location + range.length, count: objects.count),
+        "Array index out of range")
+
+      if objects.isEmpty { return }
+
+      // These objects are "returned" at +0, so treat them as pointer values to
+      // avoid retains. Copy bytes via a raw pointer to circumvent reference
+      // counting while correctly aliasing with all other pointer types.
```

Comment:
Ew. I'm pretty sure you formally want `aBuffer` to be `AutoreleasingUnsafeMutablePointer<AnyObject>`, but then you still need to avoid the copy. Rebinding to `Unmanaged<AnyObject>` would be a lot more principled.

Reply:
Do we need an explicit rebind operation to go from `UMP<AnyObject>` to `UMP<Unmanaged<AnyObject>>`? 

If so, we should revise `AutoreleasingUMP`. (In #26760 I assumed that Unmanaged<T> and T itself are in the same equivalence class.)

Reply:
@atrick, @gottesmm, and I discussed this and decided that they *could* be in the same equivalence class but probably aren't today. In this particular case, though, an explicit rebind isn't exactly hard to do.

Reply:
I think we also concluded that #26760 is based on a bad diagnosis of what happens when you *load* from a pointer. You might still prefer the new formulation of the code, but the old way wasn't in itself wrong.

Reply:
Since `UMP<T>` has a member of `T`, they are considered related types. There's no way we can disambiguate a `UP<UMP<T>>` from `UP<T>` based on their types. So, you don't strictly need the rebind as long as you add a comment that you're making use of this assumption.

Also, this case is important enough that it's worthy of documentation and comments in places to make sure we never violate it. I have that on my near-term TODO list.

Reply:
`UMP<T>` doesn't have a member of `T`, though.

Reply:
@jrose-apple for some reason I was thinking `Unamanged` and writing`UMP`

Reply:
…and this all makes sense now. Thanks!